File: solutions_21-30/Euler_26.py
### Largest sequence of repeating numbers of n < 1000 of 1/d

import timeit
import math


start = timeit.default_timer()

def is_prime(x):
    if x == 2:
        return True
    if x % 2 == 0 or x == 1:
        return False
    for i in xrange(3, int(math.sqrt(x) + 1), 2):
        if x % i == 0:
            return False
    return True


def euler_26():
    primes_to_onethousand = [i for i in xrange(1, 1000) if is_prime(i)]
    return primes_to_onethousand[len(primes_to_onethousand) - 3]


print "Answer: %s" % euler_26()
stop = timeit.default_timer()
print "Time: %s" % str(stop - start)

File: Java/java_solutions_21-30/Euler_26.java
/*
 * Find the largest sequence of repeating numbers n < 1000 of 1/d
 */

import java.util.stream.IntStream;
import java.util.List;
import java.util.stream.Collectors;


public class Euler_26 {

    public static boolean is_prime(int n) {
        if (n == 2) {
            return true;
        } else if (n % 2 == 0 || n < 2) {
            return false;
        }
        return IntStream.range(3, (int) Math.sqrt(n) + 1)
                        .allMatch(x -> n % x != 0);
    }

    public static void main(String[] args) {
        long start = System.nanoTime();
        List<Integer> primes = IntStream.range(1, 1001)
                                        .filter(x -> is_prime(x))
                                        .boxed()
                                        .collect(Collectors.toList());
        int answer = primes.get(primes.size() - 3);
        System.out.println("Answer: " + answer);
        long stop = System.nanoTime();
        System.out.printf("Time: %.4f\n", ((float) stop - start) / 1_000_000_000);
    }
}

